File: WebAPI/WebAPI/Models/Customer_Document_ADD.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WebAPI.Models
{
    public class Customer_Document_ADD
    {
        public int company_id { get; set; }
        public int customer_id { get; set; }
        public int customer_site_id { get; set; }
        public int customer_system_id { get; set; }
        public int job_id { get; set; }
        public int security_level { get; set; }
        public string file_name { get; set; }
        public int file_size { get; set; }
        public DateTime upload_date { get; set; }
        public string document_ext { get; set; }
        public string user_code { get; set; }
        public string user_description { get; set; }
        public string reference1 { get; set; }
        public string reference2 { get; set; }
        public string reference3 { get; set; }
        public string reference4 { get; set; }
        //public byte[] file_data { get; set; }
        //[NotMapped]
        public IFormFile file_data { get; set; }
        public int document_id { get; set; }
    }
}

File: angular-src/src/app/models/customerdocumentadd.ts
export class Customer_Document_ADD {
    company_id: number;
    customer_id: number;
    customer_site_id: number;
    customer_system_id: number;
    job_id: number;
    security_level: number;
    file_name: string;
    file_size: number;
    //upload_date: Date =null;
    upload_date: Date = new Date();
    //upload_date: Date;
    document_ext: string;
    user_code: string;
    user_description: string;
    reference1: string;
    reference2: string;
    reference3: string;
    reference4: string;
    //file_data: ImageBitmap;
    file_data: File = null;
    document_id: number;
}